aggregators/converter.go

```diff
@@ -197,18 +108,19 @@ func EventToCombinedMetrics(
 	cm.ServiceMetrics = append(cm.ServiceMetrics, ksm)
 
 	var errs []error
-	for partitionID, svcInstanceMetrics := range m {
+	for partitionID, sim := range collector {
 		key := unpartitionedKey
 		key.PartitionID = partitionID
 
-		cm.ServiceMetrics[0].Metrics.ServiceInstanceMetrics[0].Metrics = svcInstanceMetrics
+		cm.ServiceMetrics[0].Metrics.ServiceInstanceMetrics[0].Metrics = sim
 		cm.EventsTotal = weightedEventsTotal
 		cm.YoungestEventTimestamp = uint64(eventTS)
 		if err := callback(key, cm); err != nil {
 			errs = append(errs, err)
 		}
 		cm.ServiceMetrics[0].Metrics.ServiceInstanceMetrics[0].Metrics = nil
-		svcInstanceMetrics.ReturnToVTPool()
+		sim.ReturnToVTPool()
+		delete(collector, partitionID)
```

Comment:
Is the delete needed? The map is discarded when the function returns isn't it?

Reply:
Yup, not needed. Removed it.